components/camera/replaypcd/replaypcd.go

```diff
@@ -26,41 +39,132 @@ func init() {
 
 // Config describes how to configure the replay camera component.
 type Config struct {
-	Source   resource.Name `json:"source,omitempty"`
-	RobotID  string        `json:"robot_id,omitempty"`
-	Interval TimeInterval  `json:"time_interval,omitempty"`
+	Source   string       `json:"source,omitempty"`
+	RobotID  string       `json:"robot_id,omitempty"`
+	Interval TimeInterval `json:"time_interval,omitempty"`
 }
 
 // TimeInterval holds the start and end time used to filter data.
 type TimeInterval struct {
-	Start time.Time `json:"start,omitempty"`
-	End   time.Time `json:"end,omitempty"`
+	Start string `json:"start,omitempty"`
+	End   string `json:"end,omitempty"`
 }
 
 // Validate checks that the config attributes are valid for a replay camera.
-func (c *Config) Validate(path string) ([]string, error) {
-	return nil, nil
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.Source == "" {
+		return nil, goutils.NewConfigValidationFieldRequiredError(path, "source")
+	}
+
+	var err error
+	var startTime time.Time
+	if cfg.Interval.Start != "" {
+		startTime, err = time.Parse(timeFormat, cfg.Interval.Start)
+		if err != nil {
+			return nil, errors.New("invalid time format for start time, use RFC3339")
+		}
+		if startTime.After(time.Now()) {
+			return nil, errors.New("invalid config, start time must be in the past")
+		}
```

Comment:
Currently, we require the timestamps from the config to be in UTC to avoid this issue. we can consider also importing the local time, however, that could lead to issues on pis and linux systems which have poorly defined timezones. Keeping everything as UTC ensures consistency between systems but can introduce more work for the user.

Reply:
Thanks for checking! Spoke with Jeremy offline and as long as the values in the config are provided in UTC, this will work, since the data capture timestamps are in UTC as well. Jeremy did some manual testing on this as well.